recipes/andreasbuhr-cppcoro/all/conanfile.py

```diff
@@ -45,68 +49,73 @@ def config_options(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            if self.settings.os == "Windows" and self.options.shared:
-                self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = "ON"
-            self._cmake.configure()
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
+        tc.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.filenames["cmake_find_package"] = "cppcoro"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "cppcoro"
-        self.cpp_info.names["cmake_find_package"] = "cppcoro"
-        self.cpp_info.names["cmake_find_package_multi"] = "cppcoro"
+        self.cpp_info.set_property("cmake_file_name", "cppcoro")
+        self.cpp_info.set_property("cmake_target_name", "cppcoro::cppcoro")
 
-        comp = self.cpp_info.components["cppcoro"]
-        comp.names["cmake_find_package"] = "cppcoro"
-        comp.names["cmake_find_package_multi"] = "cppcoro"
-        comp.libs = ["cppcoro"]
+        self.cpp_info.libs = ["cppcoro"]
 
-        if self.settings.os == "Linux" and self.options.shared:
-            comp.system_libs = ["pthread"]
-        if self.settings.os == "Windows":
-            comp.system_libs = ["synchronization"]
+        if self.settings.os in ["Linux", "FreeBSD"] and self.options.shared:
+            self.cpp_info.system_libs = ["pthread", "m"]
+        elif self.settings.os == "Windows":
+            self.cpp_info.system_libs = ["synchronization", "mswsock", "ws2_32"]
 
-        if self.settings.compiler == "Visual Studio":
-            comp.cxxflags.append("/await")
+        if is_msvc(self):
+            self.cpp_info.cxxflags.append("/await")
         elif self.settings.compiler == "gcc":
-            comp.cxxflags.append("-fcoroutines")
-            comp.defines.append("CPPCORO_COMPILER_SUPPORTS_SYMMETRIC_TRANSFER=1")
-        elif self.settings.compiler == "clang" or self.settings.compiler == "apple-clang":
-            comp.cxxflags.append("-fcoroutines-ts")
+            self.cpp_info.cxxflags.append("-fcoroutines")
+            self.cpp_info.defines.append("CPPCORO_COMPILER_SUPPORTS_SYMMETRIC_TRANSFER=1")
+        elif self.settings.compiler in ["clang", "apple-clang"]:
+            self.cpp_info.cxxflags.append("-fcoroutines-ts")
```

Comment:
This flag is no longer valid in newer clang versions, should check its version beforhand